x-pack/plugins/security_solution/public/notes/store/notes.slice.ts

```diff
@@ -276,10 +276,32 @@ export const selectFetchNotesError = (state: State) => state.notes.error.fetchNo
 export const selectFetchNotesStatus = (state: State) => state.notes.status.fetchNotes;
 
 export const selectNotesByDocumentId = createSelector(
-  [selectAllNotes, (state, documentId) => documentId],
+  [selectAllNotes, (state: State, documentId: string) => documentId],
   (notes, documentId) => notes.filter((note) => note.eventId === documentId)
 );
 
+export const selectSortedNotesByDocumentId = createSelector(
```

Comment:
shouldn't we sort by the createdBy field and not the document id? document id is random I think, sorting by the createdBy timestamp is closer to the desired activity timeline behavior I think.

Reply:
hmmm I'm a bit confused. My code basically filters all the notes for the `documentId`

```typescript
.filter((note: Note) => note.eventId === documentId)
```

like the previous version of that selector was doing, but then after that sorts by the `field` and `direction`

```typescript
const { field, direction } = sort;
{...}
.sort((a: Note, b: Note) =>
    // @ts-ignore Object is possibly null or undefined
    direction === 'asc' ? (a[field] > b[field] ? 1 : -1) : a[field] > b[field] ? -1 : 1
);
```

Is that logic wrong?